algorithms/SumPrimes.js

```diff
@@ -0,0 +1,34 @@
+// Sum all the prime numbers up to and including the provided number.
+//
+// A prime number is defined as a number greater than one and having only two divisors, one and itself. For example, 2 is a prime number because it's only divisible by one and two.
+//
+// The provided number may not be a prime.
+
+
+var reducer = function(a, b) {
+  return a + b;
+};
+
+function isPrime(value) {
+    for(var i = 2; i < value; i++) {
+        if(value % i === 0) {
+            return false;
+        }
+    }
+    return value > 1;
+}
+
+function sumPrimes(value) {
+  var primesArray = [];
+  var fullArray = [];
+  for (var i = 0; i <= value; i++) {
+    fullArray.push(i);
+  }
+  for (var i = 0; i <= fullArray.length; i++) {
+    if (isPrime(fullArray[i]) === true) {
+      primesArray.push(fullArray[i]);
+    }
+  }
+
```

Comment:
It looks like `fullArray` is always equal to `i` so you can probably get rid of it and just use `i` wherever you use `fullArray` and `value + 1` instead of `fullArray.length`.